packages/babel-parser/test/fixtures/typescript/class/optional-generator/input.ts

```
@@ -0,0 +1,9 @@
+class A {
+  *[a?.b]?() { };
```

Should we report a recoverable error in cases like this, similarly to how report it for `readonly` and `declare`?

---

Actually, this is valid TypeScript syntax!.

Do you mean `readonly *[a?.b]?() { };` ?

---

Oh I meant with just `?`, I didn't realize it's actually valid syntax.